orangecontrib/text/widgets/utils/owwidget.py

```diff
@@ -0,0 +1,88 @@
+import threading
+from functools import wraps
+from PyQt4.QtCore import pyqtSlot as Slot, QMetaObject, Qt, Q_ARG
+
+from Orange.widgets.widget import OWWidget
+
+
+class StopExecution(Exception):
+    pass
+
+
+class OWConcurrentWidget(OWWidget):
```

Comment:
I thought that `@chunkable` was nice, but this ... this is the real elegance 😉 👍 👏